modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DistributedTransaction.java

```diff
@@ -238,6 +239,10 @@ public void enqueueMessage(AndesMessage andesMessage, AndesChannel andesChannel)
 
             if (totalMessageSize < maxTotalMessageSizeAllowed) {
                 branch.enqueueMessage(andesMessage);
+                if (MessageTracer.isEnabled()) {
```

Comment:
To avoid additional calls to andesMessage.getMetadata(), branch.getXid(), branch.getState() when not in tracing mode. There will be only a MessageTracer.isEnabled() call. 

If that doesn't make sense, can be removed :)